src/sync/core-sync-state.js

```diff
@@ -204,7 +204,17 @@ export class CoreSyncState {
   /**
    * @param {PeerId} peerId
    */
-  #getPeerState(peerId) {
+  disconnectPeer(peerId) {
+    const wasRemoved = this.#remoteStates.delete(peerId)
+    if (wasRemoved) {
+      this.#update()
+    }
+  }
+
+  /**
+   * @param {PeerId} peerId
+   */
+  #getOrCreatePeerState(peerId) {
```

Comment:
The name of this method was misleading, and its use in #onPeerRemove was causing a bug, because it was recreating a removed state.